Sources/cache/CacheApiInterface.php

```diff
@@ -0,0 +1,67 @@
+<?php
+
+/**
+ * Simple Machines Forum (SMF)
+ *
+ * @package SMF
+ * @author Simple Machines https://www.simplemachines.org
+ * @copyright 2020 Simple Machines and individual contributors
+ * @license https://www.simplemachines.org/about/smf/license.php BSD
+ *
+ * @version 2.1 RC2
+ */
+
+if (!defined('SMF'))
+	die('No direct access...');
+
+
+interface CacheApiInterface
+{
+	/**
+	 * Checks whether we can use the cache method performed by this API.
+	 *
+	 * @access public
+	 * @param boolean $test Test if this is supported or enabled.
+	 * @return boolean Whether or not the cache is supported
+	 */
+	public function isSupported($test = false);
```

Comment:
More of a thought here, but we do type casting in other places to be compliant with other parts of SMF.  We should make the Cache API use type casting to the full extent.

Reply:
mmm you mean type hinting?  or what do you mean by type casting?

Reply:
Yea I meant type hints.  Nothing more than a comment we should clean it up to go full type hinting if we are refactoring it to ensure proper compatibility with the API and no assumptions about how a method works or takes input.

I believe our min is still in PHP 5.x, which doesn't support return type declaration.  So we can't implant those unless we are moving the min to PHP 7.

Reply:
yep unfortunately we can't use them :( this is as far as we can enforce the interface.